### PR TITLE
Feat/add tls support

### PR DESCRIPTION
**Pull request by getlarge**
Rework #8 

- Create a separate helper to extract TLS details
- Integrate helper when proxy should be considered or not
- Refine types
- Add tests for TLS over TCP and HTTP over TLS
- Update dependencies

### CODE REVIEW

**Comment by getlarge** [comment]
@robertsLando The end of the week became the end of the month but it's done now!

File: tls/client-1-crt.pem [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-----BEGIN CERTIFICATE-----
```
Comment (GavinDmello) [inline tls/client-1-crt.pem:1]: We could move the fixtures and tests to a test folder.

Comment (getlarge) [reply tls/client-1-crt.pem:1]: You're right

**Review by robertsLando — APPROVED** [review]
LGTM 🚀

**Review by GavinDmello — APPROVED** [review]

**Comment by robertsLando** [comment]
Feel free to squash&merge and do a minor release :)

**Pull request merged**